packages/components/src/utils/config-values.js

```diff
@@ -48,7 +48,7 @@ export default {
 	borderWidth: '1px',
 	borderWidthFocus: '1.5px',
 	borderWidthTab: '4px',
-	spinnerSize: '18px',
+	spinnerSize: Number( '18' ),
```

Comment:
@ciampo let me know if this is correct.

Reply:
You should be able to just use the value `18`, without wrapping it in `Number()`. You could then also update the `stroke-dasharray` attribute, since it doesn't require a runtime `calc` anymore

<details>

<summary>For example (click to expand):</summary>

```diff
diff --git a/packages/components/src/spinner/index.js b/packages/components/src/spinner/index.js
index b256e44a75..f30a852403 100644
--- a/packages/components/src/spinner/index.js
+++ b/packages/components/src/spinner/index.js
@@ -39,8 +39,7 @@ export const SpinnerIndicator = styled.circle`
 	stroke-linecap: round;
 	stroke-width: 1.5px;
 	transform-origin: 50% 50%;
-	stroke-dasharray: ${ CONFIG.spinnerSize },
-		calc( ${ CONFIG.spinnerSize } * 10 );
+	stroke-dasharray: ${ CONFIG.spinnerSize }, ${ CONFIG.spinnerSize * 10 };
 `;
 
 export default function Spinner() {
diff --git a/packages/components/src/utils/config-values.js b/packages/components/src/utils/config-values.js
index 526f85c457..7ba2b9d3dc 100644
--- a/packages/components/src/utils/config-values.js
+++ b/packages/components/src/utils/config-values.js
@@ -48,7 +48,7 @@ export default {
 	borderWidth: '1px',
 	borderWidthFocus: '1.5px',
 	borderWidthTab: '4px',
-	spinnerSize: Number( '18' ),
+	spinnerSize: 18,
 	fontSize: '13px',
 	fontSizeH1: 'calc(2.44 * 13px)',
 	fontSizeH2: 'calc(1.95 * 13px)',
```
</details>

Reply:
I get an error when I try it this way 🤔 

<img width="876" alt="Screenshot 2022-01-20 at 12 46 21" src="https://user-images.githubusercontent.com/846565/150341268-3ff0db1e-b9da-4c82-be3e-2cbe9fe46060.png">

Thinking about this a little more, do we actually need the `spinnerSize` config value? It doesn't seem to serve much of a purpose right now, especially as all Spinner instances are currently the same size. Might a better approach be to allow a `size` prop on the `Spinner` component? That way it's easy to use different sized spinners ad hoc.

Reply:
> I get an error when I try it this way 🤔

It seems like your IDE thinks that `spinnerSize` is a `string` — did you also update `spinnerSize` in the config to `18` (as part of the diff above)?

> Thinking about this a little more, do we actually need the spinnerSize config value? It doesn't seem to serve much of a purpose right now, especially as all Spinner instances are currently the same size. Might a better approach be to allow a size prop on the Spinner component? That way it's easy to use different sized spinners ad hoc.

Let's leave it in the config for now, as it matches [the `$spinner-size` variable in the base styles](https://github.com/WordPress/gutenberg/blob/eb8c7690b4cacab028dd74e971a2e000bf5147d6/packages/base-styles/_variables.scss#L58). We have plans later this year to take a closer look at the overall config variables :) 

Re. adding a `size` prop, let's investigate that separately from this PR, maybe while we also address some of the ideas discussed in https://github.com/WordPress/gutenberg/pull/37551#discussion_r787987574

Reply:
> did you also update spinnerSize in the config (as part of the diff above)? i.e

Derp, I did not :)

> Re. adding a size prop, let's investigate that separately from this PR

Okay! I think we'll need to do _something_, because atm if you use a `Spinner` and resize it with css alone it won't scale correctly:

<img width="234" alt="Screenshot 2022-01-20 at 13 26 18" src="https://user-images.githubusercontent.com/846565/150347718-af58237a-0cc2-4e32-9604-3575d4afdd00.png">

Reply:
Oh, I see. Would that be considered a regression introduced by this PR ? Do we have instances in which the `Spinner` is rendered at a bigger size from its default?

It would still be good to add a Storybook example for this case scenario — (either in this PR or in a separate PR)

Reply:
Alright, I went ahead and had some fun trying out a couple of different approaches — I landed on what seems like a good solution, which avoids using `stroke-dasharray` and therefore scales better when making the spinner bigger (without adding a `size` prop).

These are the steps I went through:

1. Use a `circle` to render the gray "track" (instead of `box-shadow`)
2. Render the indicator (the blue arc) by using the [`path` element](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#arcs) (instead of `stroke-dasharray`)
3. Clean-up the CSS code by
  - removing the `box-shadow`, `border-radius` and `stroke-dasharray` CSS rules
  - refactoring some common styles

In order to showcase these changes in Storybook, I went ahead and allowed the `Spinner` component to receive `props` (including a classname). Finally, I added 2 more spinners to the Storybook example:

<img width="343" alt="Screenshot 2022-01-20 at 19 59 02" src="https://user-images.githubusercontent.com/1083581/150404331-d8351268-993b-4753-b567-782e99b1e3ff.png">

<details>

<summary>Click to show code changes</summary>

```diff
diff --git a/packages/components/src/spinner/index.js b/packages/components/src/spinner/index.js
index 3a28c732d8..7ea6a41b91 100644
--- a/packages/components/src/spinner/index.js
+++ b/packages/components/src/spinner/index.js
@@ -1,23 +1,47 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
 /**
  * Internal dependencies
  */
-import { StyledSpinner, SpinnerIndicator } from './styles';
+import { StyledSpinner, SpinnerTrack, SpinnerIndicator } from './styles';
 
-export default function Spinner() {
+/**
+ * @typedef OwnProps
+ *
+ * @property {string} [className] Class name
+ */
+/** @typedef {import('react').ComponentPropsWithoutRef<'svg'> & OwnProps} Props */
+
+/**
+ * @param {Props} props
+ * @return {JSX.Element} Element
+ */
+export default function Spinner( { className, ...props } ) {
 	return (
 		<StyledSpinner
-			className="components-spinner"
+			{ ...props }
+			className={ classNames( 'components-spinner', className ) }
 			viewBox="0 0 100 100"
 			xmlns="http://www.w3.org/2000/svg"
 			role="presentation"
 			focusable="false"
 		>
-			<SpinnerIndicator
-				cx="50%"
-				cy="50%"
+			{ /* Gray circular background */ }
+			<SpinnerTrack
+				cx="50"
+				cy="50"
 				r="50"
 				vectorEffect="non-scaling-stroke"
 			/>
+
+			{ /* Theme-colored arc */ }
+			<SpinnerIndicator
+				d="m 50 0 a 50 50 0 0 1 50 50"
+				vectorEffect="non-scaling-stroke"
+			/>
 		</StyledSpinner>
 	);
 }
diff --git a/packages/components/src/spinner/stories/index.js b/packages/components/src/spinner/stories/index.js
index 12b60f457b..69ab2694a2 100644
--- a/packages/components/src/spinner/stories/index.js
+++ b/packages/components/src/spinner/stories/index.js
@@ -1,10 +1,34 @@
+/**
+ * External dependencies
+ */
+import { css } from '@emotion/react';
+
 /**
  * Internal dependencies
  */
+import { useCx } from '../../utils';
 import Spinner from '../';
 
 export default { title: 'Components/Spinner', component: Spinner };
 
 export const _default = () => {
-	return <Spinner />;
+	const cx = useCx();
+	const mediumSpinner = cx( css`
+		width: 100px;
+		height: 100px;
+	` );
+	const largeSpinner = cx( css`
+		width: 200px;
+		height: 200px;
+	` );
+	return (
+		<>
+			<Spinner />
+			<br />
+			<Spinner className={ mediumSpinner } />
+			<br />
+			<Spinner className={ largeSpinner } />
+			<br />
+		</>
+	);
 };
diff --git a/packages/components/src/spinner/styles.js b/packages/components/src/spinner/styles.js
index 0d2e4c0761..4a6cb4ee70 100644
--- a/packages/components/src/spinner/styles.js
+++ b/packages/components/src/spinner/styles.js
@@ -2,7 +2,7 @@
  * External dependencies
  */
 import styled from '@emotion/styled';
-import { keyframes } from '@emotion/react';
+import { css, keyframes } from '@emotion/react';
 
 /**
  * Internal dependencies
@@ -10,22 +10,19 @@ import { keyframes } from '@emotion/react';
 import { COLORS, CONFIG } from '../utils';
 
 const spinAnimation = keyframes`
-     from {
-         transform: rotate(0deg);
-     }
-     to {
-         transform: rotate(360deg);
-     }
+	from {
+		transform: rotate(0deg);
+	}
+	to {
+		transform: rotate(360deg);
+	}
  `;
 
 // The Circle can only have a centered stroke so a stacked box-shadow on the svg accomplishes a matching 1.5px border
 export const StyledSpinner = styled.svg`
 	width: ${ CONFIG.spinnerSize }px;
 	height: ${ CONFIG.spinnerSize }px;
-	box-shadow: inset 0 0 0 0.75px ${ COLORS.gray[ 300 ] },
-		0 0 0 0.75px ${ COLORS.gray[ 300 ] };
 	display: block;
-	border-radius: 50%;
 	margin: auto;
 	position: relative;
 	color: var( --wp-admin-theme-color );
@@ -33,11 +30,19 @@ export const StyledSpinner = styled.svg`
 	animation: 1.4s linear infinite both ${ spinAnimation };
 `;
 
-export const SpinnerIndicator = styled.circle`
+const commonPathProps = css`
 	fill: transparent;
+	stroke-width: 1.5px;
+`;
+
+export const SpinnerTrack = styled.circle`
+	${ commonPathProps };
+	stroke: ${ COLORS.gray[ 300 ] };
+`;
+
+export const SpinnerIndicator = styled.path`
+	${ commonPathProps };
 	stroke: currentColor;
 	stroke-linecap: round;
-	stroke-width: 1.5px;
 	transform-origin: 50% 50%;
-	stroke-dasharray: ${ CONFIG.spinnerSize }, ${ CONFIG.spinnerSize * 10 };
 `;
```
</summary>

I think this approach could solve a couple of problems without introducing much complexity, what do you think?

Reply:
Oh nice, this looks more comprehensive – thanks so much :) Please feel free to push the changes.

Do you think it's worth including the preset sizes in the component as a part of this PR? If so the only tweak I would suggest would be to use multiples of our grid unit (or spinner-size) values for the medium/large spinners. 

Now that I think of it, it might be a good idea to make the default `spinnerSize` 16px so that it better aligns with the broader grid.

Reply:
> Please feel free to push the changes.

Thank you ! I went ahead and pushed those changes (including a change to the README to remove the out of date screenshot).

> Do you think it's worth including the preset sizes in the component as a part of this PR? If so the only tweak I would suggest would be to use multiples of our grid unit (or spinner-size) values for the medium/large spinners.

I did include the updates to the Storybook examples, if anything to test that the stroke width is not changing as we increase the overall size of the spinner — I also made sure that those storybook examples use multiples of the grid unit.

For now, I wouldn't include any new APIs to the component (e.g. `size`), since the previous one also didn't have it.

> Now that I think of it, it might be a good idea to make the default spinnerSize 16px so that it better aligns with the broader grid.

I haven't made that change yet as I'm not sure if resizing the spinner (from `18px` to `16px`) would have any ripercussions / would be considered breaking change. @jasmussen do you have any opinions on this?

Reply:
Thanks, this is looking fantastic :) 

I think there's a reasonable case to resize from 18px to 16px. It aligns with our grid unit, and most other icons tend to render at ~16px, e.g. Styles icon:

<img width="170" alt="Screenshot 2022-01-21 at 16 36 14" src="https://user-images.githubusercontent.com/846565/150565004-ea98b958-dee4-435a-99e3-fedaca659c00.png">

Not a strong feeling though, and happy to defer to @jasmussen on this point.

Reply:
I like it.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/37551/commits/91a861fa30bc0a8ebc957027214d3952c7d79796

Let me know once you've had a look, I think we may be able to merge this PR :)